scipy/optimize/tests/test__differential_evolution.py

```diff
@@ -554,3 +559,576 @@ def test_converged(self):
+                    x[3]*x[8] - 1250000 - x[3]*x[5] + 2500*x[5]]
+
+        L = LinearConstraint(A, -np.inf, 1)
+        N = NonlinearConstraint(c1, 0, np.inf)
+
+        bounds = [(100, 10000)] + [(1000, 10000)]*2 + [(10, 1000)]*5
+        constraints = (L, N)
+
+        with suppress_warnings() as sup:
+            sup.filter(UserWarning)
+            res = differential_evolution(f, bounds, strategy='rand1bin',
+                                     seed=1234, constraints=constraints,
+                                     popsize=3)
+
+        f_opt = 7049.330923
+        x_opt = (579.3167, 1359.943, 5110.071, 182.0174, 295.5985, 217.9799,
+                 286.4162, 395.5979)
+
+        assert_allclose(f(x_opt), f_opt)
+        assert_allclose(res.fun, f_opt, atol=0.1)
+        assert_allclose(res.x, x_opt, atol=0.12)
+        assert res.success
+        assert_(np.all(A @ res.x <= b))
+        assert_(np.all(np.array(c1(res.x)) >= 0))
+        assert_(np.all(res.x >= np.array(bounds)[:, 0]))
+        assert_(np.all(res.x <= np.array(bounds)[:, 1]))
+
+    def test_L5(self):
+        # Lampinen ([5]) test problem 5
+
+        def f(x):
+            x = np.hstack(([0], x))  # 1-indexed to match reference
+            fun = (np.sin(2*np.pi*x[1])**3*np.sin(2*np.pi*x[2]) /
+                   (x[1]**3*(x[1]*x[2])))
+            return -fun  # maximize
+
+        def c1(x):
+            x = np.hstack(([0], x))  # 1-indexed to match reference
+            return [x[1]**2 - x[2] + 1,
+                    1 - x[1] + (x[2]-4)**2]
+
+        N = NonlinearConstraint(c1, -np.inf, 0)
+        bounds = [(0, 10)]*2
+        constraints = (N)
+
+        res = differential_evolution(f, bounds, strategy='rand1bin', seed=1234,
+                                     constraints=constraints)
+
+        x_opt = (1.22067691, 4.24096915)
+        f_opt = -0.100738
+
+        assert_allclose(f(x_opt), f_opt, atol=2e-7)
+        assert_allclose(res.x, x_opt, atol=5e-3)
+        assert_allclose(res.fun, f_opt, atol=1e-4)
+        assert res.success
+        assert_(np.all(np.array(c1(res.x)) <= 0))
+        assert_(np.all(res.x >= np.array(bounds)[:, 0]))
+        assert_(np.all(res.x <= np.array(bounds)[:, 1]))
+
+    def test_L6(self):
+        # Lampinen ([5]) test problem 6
+        def f(x):
+            x = np.hstack(([0], x))  # 1-indexed to match reference
+            fun = (x[1]-10)**3 + (x[2] - 20)**3
+            return fun
+
+        def c1(x):
+            x = np.hstack(([0], x))  # 1-indexed to match reference
+            return [(x[1]-5)**2 + (x[2] - 5)**2 - 100,
+                    -(x[1]-6)**2 - (x[2] - 5)**2 + 82.81]
+
+        N = NonlinearConstraint(c1, 0, np.inf)
+        bounds = [(13, 100), (0, 100)]
+        constraints = (N)
+        res = differential_evolution(f, bounds, strategy='rand1bin', seed=1234,
+                                     constraints=constraints)
+        x_opt = (14.095, 0.84296)
+        f_opt = -6961.814744
```

Comment:
I was playing around with this one because the tolerance was not very tight. `slsqp` found the optimum reported by Lampinen. Funny, though - I tried manually polishing with `trust-constr` and got a different value than the auto-polishing by DE. Thoughts on why that might happen?

Reply:
If the starting points were the same (unpolished DE), then I don't know why.

Reply:
I can get to the best minimum if I set the de tolerance to be very low. As you said, the polishing done in DE isn't getting there. The DE polish is only accepted if the polish is successful, the solution lies strictly within the bounds, and polish.fun is less than de.fun. Perhaps that's why the polishing isn't getting there.